examples/cpp/modes/goto/include/mode.hpp

```diff
@@ -0,0 +1,222 @@
+            // stop caring about heading (the arctangent becomes undefined)
+            _goto_setpoint->update(target_position_m);
+          } else {
+            _goto_setpoint->update(target_position_m, heading_target_rad);
+          }
+
+          if (positionReached(target_position_m)) {
+            _state = State::kGoingEast;
+          }
+        }
+        break;
+
+      case State::kGoingEast: {
+          // go to the northeast corner while spinning
+          const Eigen::Vector3f target_position_m = _start_position_m +
+            Eigen::Vector3f{kTriangleHeight, kTriangleWidth, 0.F};
+
+          // scale the speed limits by distance to the target
+          const Eigen::Vector2f vehicle_to_target_xy = target_position_m.head(2) -
+            _vehicle_local_position->position().head(2);
+          const float speed_scale = std::min(vehicle_to_target_xy.norm() / kTriangleWidth, 1.F);
+
+          const float max_horizontal_velocity_m_s = 5.F * speed_scale + (1.F - speed_scale) * 1.F;
+          const float max_vertical_velocity_m_s = 3.F * speed_scale + (1.F - speed_scale) * 0.5F;
+          const float max_heading_rate_rad_s = (45.F * speed_scale + (1.F - speed_scale) * 25.F) *
+            (float)M_PI / 180.F;
+          const float heading_setpoint_rate_of_change =
+            (40.F * speed_scale + (1.F - speed_scale) * 20.F) * (float)M_PI / 180.F;
+
+          if (!_start_heading_set) {
+            _spinning_heading_rad = _vehicle_heading_rad;
+            _start_heading_set = true;
+          }
+
+          if (!positionReached(target_position_m)) {
+            _spinning_heading_rad += heading_setpoint_rate_of_change * dt_s;
+          }
+
+          _goto_setpoint->update(
+            target_position_m,
+            _spinning_heading_rad,
+            max_horizontal_velocity_m_s,
+            max_vertical_velocity_m_s,
+            max_heading_rate_rad_s);
+
+          if (positionReached(target_position_m)) {
+            _state = State::kGoingSouthwest;
+          }
+        }
+        break;
+
+      case State::kGoingSouthwest: {
+          // go to southwest corner while facing the northeastern corner
+          const Eigen::Vector2f position_of_interest_m = _start_position_m.head(2) +
+            Eigen::Vector2f{kTriangleHeight, 0.F};
+          const Eigen::Vector2f vehicle_to_poi_xy = position_of_interest_m -
+            _vehicle_local_position->position().head(2);
+          const float heading_target_rad = atan2f(vehicle_to_poi_xy(1), vehicle_to_poi_xy(0));
+
+          _goto_setpoint->update(_start_position_m, heading_target_rad);
+          if (positionReached(_start_position_m)) {
+            completed(px4_ros2::Result::Success);
+            return;
+          }
+        }
+        break;
+    }
+  }
+
+private:
+  static constexpr float kTriangleHeight = 20.F; // [m]
+  static constexpr float kTriangleWidth = 30.F; // [m]
+
+  enum class State
+  {
+    kSettlingAtStart = 0,
```

Comment:
enum's don't use the constants naming
```suggestion
    SettlingAtStart = 0,
```

Reply:
done